src/pfe/portal/modules/MetricsService.js

```diff
@@ -214,129 +250,210 @@ function getNewContentsOfPomXml(originalContents) {
+  if (agentExeNeeded) {
+    newExecutionArray.push(
+      {
+        id: [ 'copy-javametrics-agent' ],
+        phase: [ 'package' ],
+        goals: [ { goal: [ 'copy-dependencies' ] } ],
+        configuration: [
           {
-            id: [ 'copy-javametrics-agent' ],
-            phase: [ 'package' ],
-            goals: [ { goal: [ 'copy-dependencies' ] } ],
-            configuration: [
-              {
-                stripVersion: [ 'true' ],
-                outputDirectory: [
-                  '${project.build.directory}/liberty/wlp/usr/servers/defaultServer/resources/'
-                ],
-                includeArtifactIds: [ 'javametrics-agent' ]
-              }
-            ]
-          },
+            stripVersion: [ 'true' ],
+            outputDirectory: [
+              '${project.build.directory}/liberty/wlp/usr/servers/defaultServer/resources/'
+            ],
+            includeArtifactIds: [ 'javametrics-agent' ]
+          }
+        ]
+      }
+    );
+  }
+  if (asmExeNeeded) {
+    newExecutionArray.push(
+      {
+        id: [ 'copy-javametrics-asm' ],
+        phase: [ 'package' ],
+        goals: [ { goal: [ 'copy-dependencies' ] } ],
+        configuration: [
           {
-            id: [ 'copy-javametrics-asm' ],
-            phase: [ 'package' ],
-            goals: [ { goal: [ 'copy-dependencies' ] } ],
-            configuration: [
-              {
-                outputDirectory: [
-                  '${project.build.directory}/liberty/wlp/usr/servers/defaultServer/resources/asm'
-                ],
-                includeGroupIds: [ 'org.ow2.asm' ],
-                includeArtifactIds: [ 'asm,asm-commons' ]
-              }
-            ]
+            outputDirectory: [
+              '${project.build.directory}/liberty/wlp/usr/servers/defaultServer/resources/asm'
+            ],
+            includeGroupIds: [ 'org.ow2.asm' ],
+            includeArtifactIds: [ 'asm,asm-commons' ]
           }
         ]
       }
-    ]
-  };
-  const newBuildPluginExecutions = originalBuildPlugins.concat(metricsCollectorBuildPlugin);
-  return newBuildPluginExecutions;
+    );
+  }
+  if (mavenDepPluginNeeded) {
+    const metricsCollectorBuildPlugin = {
+      groupId: [ 'org.apache.maven.plugins' ],
+      artifactId: [ 'maven-dependency-plugin' ],
+      version: [ '3.0.1' ],
+      executions: [
+        {
+          execution: newExecutionArray,
+        }
+      ]
+    };
+    const newBuildPlugins = originalBuildPlugins.concat(metricsCollectorBuildPlugin);
+    return newBuildPlugins;
+  }
+  
+  
+  let metricsCollectorBuildPlugin = originalBuildPlugins[mavenDepPluginIndex];
+  if (metricsCollectorBuildPlugin.executions) {
+    metricsCollectorBuildPlugin.executions[0].execution = metricsCollectorBuildPlugin.executions[0].execution.concat(newExecutionArray);
+  } else {
+    metricsCollectorBuildPlugin.executions = [ { execution: newExecutionArray } ];
+  }
+  let newBuildPlugins = originalBuildPlugins;
```

Comment:
overwriting `originalBuildPlugins` is dangeous, we could do `const newBuildPlugins = deepCopy(originalBuildPlugins)`?